tests/fixtures/tables/auto_sales.rs

```diff
@@ -0,0 +1,614 @@
+// Copyright (c) 2023-2024 Retake, Inc.
```

Comment:
This file is huge. Do we really need an entire new fixture or can we reuse our existing ones with `nyc_trips`? If not, is there any way to shorten this file?

Overall I'm quite happy with this PR, but I just find it to introduce a lot of new testing utilities. Some of them are also built in this file, but don't seem to be actual tables. These utilities should probably be moved to a separate file in `tests/fixtures/`. Perhaps something like `parquet.rs` for the batch writing, or `s3.rs` or something like that.

@Weijun-H Would love your help thoroughly reviewing this PR. Ming is busy with search-related work.

Once we clean up this file, I am happy to approve and merge this. Thank you for your patience @shamb0.

Reply:
Tl;dr:

- If we really need a full separate table, let's keep only that in this file and move the utilities to a properly scoped. file
- If we don't need a full separate table, let's use the NYC trips test fixture

Thank you!